### PR TITLE
fixed history fetching when period='max'

### PR DESCRIPTION
**Pull request by SnowCheetos**
#1916 Fixed limited period auto adjust with "max" period value

accidentally misspelled 'period' for branch name...

### CODE REVIEW

**Comment by ValueRaider** [comment]
I haven't tested but 2 considerations:
- floor vs ceil
- UTC vs exchange timezone

File: yfinance/scrapers/history.py [hunk 1]
```diff
@@ -98,6 +99,14 @@ def history(self, period="1mo", interval="1d",
...
+                
+                elif interval == "1h":
+                    start = end - 63072000 # Subtract 730 days
+                
+                # 5m to 90m interval supports up to 60 days
+                elif interval in ("5m", "15m", "30m", "60m", "90m"):
+                    start = end - 5184000 # Subtract 60 days
+
                 else:
                     max_start_datetime = pd.Timestamp.utcnow().floor("D") - _datetime.timedelta(days=99 * 365)
```
Comment (SnowCheetos) [inline yfinance/scrapers/history.py:111]: This was the current implementation on the main branch, left it as is

File: yfinance/scrapers/history.py [hunk 2]
```diff
@@ -98,6 +99,14 @@ def history(self, period="1mo", interval="1d",
             if start is None:
                 if interval == "1m":
                     start = end - 604800  # Subtract 7 days
+                
+                elif interval == "1h":
```
Comment (SnowCheetos) [inline yfinance/scrapers/history.py:103]: 1h can fetch up to 730 days of data, subtracted 730 * 24 * 60 * 60 = 63,072,000 seconds

File: yfinance/scrapers/history.py [hunk 3]
```diff
@@ -98,6 +99,14 @@ def history(self, period="1mo", interval="1d",
...
+                    start = end - 63072000 # Subtract 730 days
+                
+                # 5m to 90m interval supports up to 60 days
+                elif interval in ("5m", "15m", "30m", "60m", "90m"):
```
Comment (SnowCheetos) [inline yfinance/scrapers/history.py:107]: I tested these intervals with the main branch, and they all support up to 60 days, including 60m interval which should be equivalent to 1h, unless there is another reason for it?

Comment (ValueRaider) [reply yfinance/scrapers/history.py:107]: Last I checked, Yahoo treats `1h` different, which could be different to `60m`.

Comment (ValueRaider) [reply yfinance/scrapers/history.py]: 60m is equivalent to 1h, so I've fixed.

**Pull request merged**